java/vector/src/main/codegen/templates/UnionFixedSizeListWriter.java

```diff
@@ -183,7 +183,7 @@ public void write(DecimalHolder holder) {
 
   @Override
   public void writeNull() {
-    writer.setPosition(writer.idx()+1);
+    writer.writeNull();
```

Comment:
can you please add a check here

if (writer.idx() >= (idx() + 1) * listSize) {
throw new IllegalStateException(String.format("values at index %s is greater than listSize %s", idx(), listSize));
}

Reply:
done